core/src/server/rpc_module.rs

```diff
@@ -729,8 +740,8 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 pub struct SubscriptionSink {
 	/// Sink.
 	inner: MethodSink,
-	/// Close
-	close: async_channel::Receiver<()>,
+	/// Get notified when subscribers leave so we can exit
+	close_notify: Option<Arc<Notify>>,
```

Comment:
does this have to be an `Option`?

might get rid of `Arc::clone` but would make the code must harder to reason about...

Reply:
The only reason it's an `Option` is for the `SubscriptionSink` to be able to [check if we've been notified](https://github.com/paritytech/jsonrpsee/blob/fa2072fe28e66385d3fb981d50363ed76b77b741/core/src/server/rpc_module.rs#L831) about the Subscriber leaving. A `Notify` doesn't have a `was_notified()` thing. Can you think of a better way?

Reply:
No :)